### PR TITLE
docs: add contract addr references and seq diagram

### PR DESCRIPTION
**Pull request by nichonien**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,10 +4,55 @@
 
 # Node.js Passport strategy using decentralised identifiers
 
-## Getting Started
+## Description
 
 This repository consists of a Node.js Password Strategy which provides verification of the issuance of credential made regarding roles defined in an Ethereum Naming System (ENS).
```
Comment (JGiter) [inline README.md:9]:
```suggestion
This repository consists of a Node.js Password Strategy which authenticates based on roles defined in Energy Web Foundation Ethereum namespace
```

File: README.md [hunk 2]
```diff
@@ -4,10 +4,55 @@
...
+    participant C as Client
+    participant LS as LoginStrategy
+    participant CV as ClaimVerifier
+    participant IV as IssuerVerification
+    participant CS as Cache Server
+    participant IPFS
+    participant DR as DomainReader
+rect rgb(200, 255, 255)
+    Note right of C: Initialisation
+    C->C: Client sets the login strategy options and initialises LoginStrategy
```
Comment (JGiter) [inline README.md:25]: Arrow is better correspond to call
```suggestion
    C->>C: Client sets the login strategy options and initialises LoginStrategy
```

File: README.md [hunk 3]
```diff
@@ -4,10 +4,55 @@
...
+## Sequence diagram
+
+ ```mermaid
+ sequenceDiagram
+autonumber
+    participant C as Client
+    participant LS as LoginStrategy
+    participant CV as ClaimVerifier
+    participant IV as IssuerVerification
+    participant CS as Cache Server
```
Comment (JGiter) [inline README.md:20]:
```suggestion
    participant CS as SSI-HUB
```

File: README.md [hunk 4]
```diff
@@ -4,10 +4,55 @@
...
+    participant DR as DomainReader
+rect rgb(200, 255, 255)
+    Note right of C: Initialisation
+    C->C: Client sets the login strategy options and initialises LoginStrategy
+    Note right of C: Validation Call
+    C->LS: LoginStrategy.validate(token, payload)
+end
+rect rgb(255, 220, 255)
+    Note right of LS: Signature and role validation
+    LS->LS: Verifies token and payload (signatures)
```
Comment (JGiter) [inline README.md:31]: I think we can specify that token signature is verified by ProofVerifier?
```suggestion
    LS->>PV: Authenticate token issuer
```

**Review by JGiter — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (nichonien) [reply README.md:31]: @JGiter What is PV?

Comment (JGiter) [reply README.md:31]: ProofVerifier from ew-did-registry

**Review by JGiter — APPROVED** [review]
It is also would be useful to mention that LoginStrategy can be configured to authenticate only specific EWF role

**Review by jrhender — CHANGES_REQUESTED** [review]
Great start @nichonien . From what I can tell no requests to IPFS are shown in the diagram. Is this the case or did I just miss those. IMO, it would be important to show those.

**Comment by nichonien** [comment]
> Great start @nichonien . From what I can tell no requests to IPFS are shown in the diagram. Is this the case or did I just miss those. IMO, it would be important to show those.

It was there earlier but then I replaced it with `CredentialResolver`, I would add it back and show as `CredentialResolver` process.

**Review by JGiter — APPROVED** [review]

**Comment by nichonien** [comment]
> Great start @nichonien . From what I can tell no requests to IPFS are shown in the diagram. Is this the case or did I just miss those. IMO, it would be important to show those.

@jrhender Does the latest version aligns with the way you see it?

**Review by jrhender — APPROVED** [review]
Yes, looks great, thanks @nichonien

**Pull request merged**